### ISSUE TITLE
Support MongoDB 4.0 Multi-document transactions

### ISSUE THREAD

**Issue by pierreis** [mongodb-haskell/mongodb#94]

**Comment by why-not-try-calmer** [comment]
I would like to see multiple-document transactions implemented too. To the best of my knowledge the last MongoDB version that didn't support transactions was 3.6, EOL'ed in 2021 (the upcoming next stable, official version is  6.0 -- https://www.mongodb.com/docs/upcoming/release-notes/6.0/#std-label-release-notes-6.0). It's hard for this library to compete with librairies for other databases without transactions.

And I would be willing to help. As far as I understand this library sends opcodes + document data as bytestrings encoded and transported as per MongoDB Wire Protocol; and Write Protocol defines transactions using the OP_MSG opcode. This suggests that implementing transactions entails enabling this library to produce "notices" written using this new opcode?

**Comment by VictorDenisov** [comment]
Yep. It means implementing operations using OP_MSG code according to https://www.mongodb.com/docs/manual/reference/mongodb-wire-protocol/

**Comment by why-not-try-calmer** [comment]
> Yep. It means implementing operations using OP_MSG code according to https://www.mongodb.com/docs/manual/reference/mongodb-wire-protocol/

Okay. I don't want to promise anything but I'll look into it. If a maintainer of this library is already working on this please let me know.
